app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -409,9 +414,8 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                 try {
                     BigDecimal balance = Utils.parseBalance(s.toString(), tempLoyaltyCard.balanceType);
                     updateTempState(LoyaltyCardField.balance, balance);
-                    validBalance = true;
                 } catch (ParseException e) {
-                    validBalance = false;
+                    balanceField.setError(getString(R.string.parsingBalanceFailed, balanceField.getText().toString()));
```

Comment:
Looking at your screenshot the message doesn't fit below the box so we definitely need to simplify this string. Maybe just remove it and say "Invalid balance"

![](https://user-images.githubusercontent.com/23402988/216807290-8a999554-aec4-4332-958a-7c5ed6420cef.jpg)

Reply:
@TheLastProject , amended.
It now looks like this:

![catima](https://user-images.githubusercontent.com/23402988/216894383-ba586bde-7b25-4106-8e35-d75d2625bbde.jpg)

Would you like to create a string entry for invalid balance in strings.xml?

Reply:
Yeah, that would be good. The old `parsingBalanceFailed` string can be remove then too as I believe this is the only place it is used.

Reply:
Done.